### PR TITLE
(maint) adapt tests for RHEL 9 Beta

### PR DESCRIPTION
**Pull request by ciprianbadescu**

### CODE REVIEW

File: acceptance/lib/facter/acceptance/base_fact_utils.rb [hunk 1]
```diff
@@ -144,7 +144,12 @@ def el_expected_facts(agent)
           os_distro_release_full = /#{os_version}\.\d+/
         else
           os_name = 'RedHat'
-          os_distro_description = /Red Hat Enterprise Linux( Server)? release #{os_version}\.\d+ \(\w+\)/
+          require 'pry-byebug'; binding.pry
```
Comment (GabrielNagy) [inline acceptance/lib/facter/acceptance/base_fact_utils.rb:147]: you probably want to remove this 😋

Comment (joshcooper) [reply acceptance/lib/facter/acceptance/base_fact_utils.rb]: Updated

**Comment by joshcooper** [comment]
Ruby 3 cell is using Ruby 3.1. Need to apply the same fix from puppet

**Comment by joshcooper** [comment]
Confirmed that the networking and dhcp test fixes aren't needed in 3.x since it only checks a subset of facts: 
```
        expected_facts = {
            'os.architecture'          => os_arch,
            'os.family'                => 'RedHat',
            'os.hardware'              => os_hardware,
            'os.name'                  => os_name,
            'os.release.full'          => /#{os_version}(\.\d+)?(\.\d+)?/,
            'os.release.major'         => os_version,
            'os.release.minor'         => /(\d+)?/,
            'processors.count'         => /[1-9]/,
            'processors.physicalcount' => /[1-9]/,
            'processors.isa'           => os_hardware,
            'processors.models'        => processor_model_pattern,
            'kernel'                   => 'Linux',
            'kernelrelease'            => /\d+\.\d+\.\d+/,
            'kernelversion'            => /\d+\.\d+/,
            'kernelmajversion'         => /\d+\.\d+/
        }
```

**Review by joshcooper — APPROVED** [review]

**Pull request merged**